plugins/org.eclipse.mat.api/src/org/eclipse/mat/inspections/LeakHunterQuery.java

```diff
@@ -590,6 +606,81 @@ else if (snapshot.isClass(accumulationPointId))
+                    throws SnapshotException
+    {
+        return getRetainedHeapTopConsumersDescription(new int[] { object }, listener, topItems);
+    }
+
+    private String getRetainedHeapTopConsumersDescription(int[] objects, IProgressListener listener, int topItems)
+                    throws SnapshotException
+    {
+        if (objects.length == 0)
+        { return Messages.LeakHunterQuery_RetainedHeapComponentEmpty; }
+        int[] minRetainedSet = snapshot.getMinRetainedSet(objects, listener);
+
+        Map<String, Long> sumHeapSizes = new HashMap<>();
+        Map<String, Integer> counts = new HashMap<>();
+        for (int objectId : minRetainedSet)
+        {
+            IObject object = snapshot.getObject(objectId);
+            String name;
+            if (snapshot.isClass(objectId))
+            {
+                IClass clazz = (IClass) object;
+                name = clazz.getName();
+            }
+            else
+            {
+                name = object.getClazz().getName();
+            }
+
+            Long sumHeapSize = sumHeapSizes.get(name);
+            if (sumHeapSize == null)
+            {
+                sumHeapSize = 0L;
+            }
+            sumHeapSize += snapshot.getHeapSize(objectId);
+            sumHeapSizes.put(name, sumHeapSize);
+
+            Integer count = counts.get(name);
+            if (count == null)
+            {
+                count = 0;
+            }
+            count++;
+            counts.put(name, count);
+        }
+        StringBuilder result = new StringBuilder();
+        List<Entry<String, Long>> list = new ArrayList<>(sumHeapSizes.entrySet());
+        list.sort((e1, e2) -> e2.getValue().compareTo(e1.getValue()));
+        List<Entry<String, Long>> finalList = list.subList(0, Math.min(list.size(), topItems));
+        for (int i = 0; i < finalList.size(); i++)
+        {
+            Entry<String, Long> entry = finalList.get(i);
+            if (result.length() > 0)
+            {
+                result.append(", "); //$NON-NLS-1$
+            }
+            if (i == finalList.size() - 1 && finalList.size() > 1)
+            {
+                result.append(Messages.LeakHunterQuery_RetainedHeapComponentAnd);
+            }
+            int count = counts.get(entry.getKey());
+            if (count == 1)
+            {
+                result.append(MessageUtil.format(Messages.LeakHunterQuery_RetainedHeapComponentInstance, entry.getKey(),
+                                count, bytesFormatter.format(entry.getValue())));
+            }
+            else
+            {
+                result.append(MessageUtil.format(Messages.LeakHunterQuery_RetainedHeapComponentInstances,
+                                entry.getKey(), count, bytesFormatter.format(entry.getValue())));
+            }
+        }
+        return HTMLUtils.escapeText(result.toString());

```

Comment:
I don't think we want all of this escaped, otherwise I see `<..>` etc in the output.

![image](https://github.com/user-attachments/assets/755c5d68-4bc7-44b0-bdbd-0aa3dd83e37d)

Reply:
Oops. I added escaping to fix some tests (classes with `<` and `>` in them) but then I forgot to run it again visually to see this.